### DISCUSSION TITLE
Thank you for keeping GLASS somewhat functional!

### DISCUSSION THREAD

**Discussion by ephsf** [tujson/GlassEcho#10] [General]
I'm sure it's just a labour of love but many thanks for making notifications functional again. I'm grateful for your work to date. 

I don't know if you plan on updating the functionality (the text to speech for notification responses sounds amazing if you can get it to work!) but thank you. I wish I could help - I only have  java and flutter/dart experience but never actually took a look at the Glass Dev Tools...

Best of luck with your endeavours and once again thank you!

**Comment by tujson** [comment]
Hi! Thanks for the kind words!

Unfortunately I haven't had much time these past few months to work on GlassEcho. I'd be more than willing to review contributions and offer guidance wherever possible. I _might_ free up some early next year to take a stab at another feature or two.

Although it's not 1:1, Flutter has quite a few concepts similar to Android (speaking from experience, but going from Android -> Flutter :)). 

I think the best way forward for this project is to talk to daily users (if they even exist!) and see what kind of pain points they're dealing with. Text-to-speech is indeed super awesome, but I'm not too sure folks are comfortable using that kind of tech yet (great place to look would be reviews for similar commercial "smart glasses", like Echo Frames).

- Is the connection process still too laborious?
- When are users interacting with notifications on Glass?
  - Or are they just dismissing them vs. using Glass as a way to filter out unimportant notifications and pulling out the phone for ones that are more pressing?
  - If they're using notification actions, which ones? Reply? "Like"ing a message?
- How do we add some more contextual relevance? Or should we (privacy concerns)?
  - I absolutely hated being spammed with text messages while having a face-to-face conversation with someone. Aside from the bone conduction speaker's semi-annoying vibration at the time, it felt super rude to quickly glance up (felt involuntary). Also felt self-conscious since I'm pretty sure the other person can see Glass light up when I get a notification. Ended up taking Glass off for the remainder of the conversation. 

I've heard quite a few people still use Glass for photos, so perhaps notifications aren't as important for them, but a process to automatically send photos from Glass to their phone would be invaluable.

And for personal curiosity, I'd be super interested to hear about how people feel while wearing Glass while going about their daily lives. I don't anymore since Seattle is a bit less tech-friendly than a college campus, but I do find myself oftentimes wishing I had them on!

++ There's also the elephant in the room, what with Glass XE being discontinued for so many years. There are quite a few alternatives (Glass EE2, Tooz, Vuzix, ...) for which GlassEcho should work with little to no modifications.
